cores/esp8266/Print.cpp

```diff
@@ -261,6 +283,28 @@ size_t Print::printNumber(unsigned long n, uint8_t base) {
         unsigned long m = n;
         n /= base;
         char c = m - base * n;
+
+        *--str = c < 10 ? c + '0' : c + 'A' - 10;
+    } while(n);
+
+    return write(str);
+}
+
+size_t Print::printNumber(unsigned long long n, uint8_t base) {
```

Comment:
I think it would be best to unify this in a single template method where the type of n is the template parameter. It will reduce mistakes. Something like:
```cpp
template <typename uintType>
size_t Print::printNumber(uintType n, uint8_t base) {
    char buf[8 * sizeof(n) + 1]; // Assumes 8-bit chars plus zero byte.
...
    do {
        uintType m = n;
        ...
    }
...
}
```

Reply:
Not really, this might require specializations anyway, maybe not on the ESP8266 right now, but if the compiler intrinsics change, I don't know about the CPU instruction set? uint32_t and uint64_t modulo are implemented differently on the other boards.

Reply:
If other boards require specializations, they can do that, or they can reimplement this, or something else. We're on the ESP8266, so we focus on the ESP8266. 
The intrinsics aren't going to change. It's a 32bit platform, and our build system is defined. Also, templatization should cover that anyways even if it does happen.